### ISSUE TITLE
investigate methods to backup the redeem script

### ISSUE THREAD

**Issue by jolan** [decred/dcrstakepool#237]
If both a stake pool and a user lose the script, there really isn't a good way to reconstruct it at the moment.

This should at least be documented until a solution is implemented.

**Comment by JoeGruffins** [comment]
I looked into this a bit. In the end you need the user's pubkey, that we don't store. So any attempt to recreate the script would require that information.
If the user can provide that key, and the multisig address they gave voting rights too, which is the redeem script's public address, we can find the address we used to make the original redeem script by looping through our known addresses creating multisigs until we find a match. This is doable for address that have already voted, but for an address that hasn't voted even once yet we would need to probe unused addresses to some index number while searching.
I'm sorry to butcher terms.

**Comment by xaur** [comment]
I believe to fundamentally fix the problem something like https://github.com/decred/dcrwallet/issues/1217 is needed.

**Comment by JoeGruffins** [comment]
I agree @xaur and that's where I will be spending my time.... When I have some.

**Comment by xaur** [comment]
@JoeGruffins time is tricky indeed, but when you realize you _want_ it, the [allocation](https://www.reddit.com/r/decred/comments/biipw3/im_going_allin_on_decred_here_is_why/em7vabu/) happens.

**Comment by cball321** [comment]
I currently have an expired ticket where the asp dropped out and I did not backup my redeem script.(yes stupid I know)  My wallet shows the expired ticket, but I do not have access to the DCR used to purchase it. Is there a way for me to retrieve my lost DCR without have the redeem script?  Any help would be appreciated

**Comment by JoeGruffins** [comment]
@cball321 I'm going to say no, but please join support on matrix:
 https://matrix.to/#/!dhHYPTtCtvPSUfTepT:decred.org?via=decred.org&via=matrix.org&via=zettaport.com
There may be something that can be done. Let me know if that link doesn't work.

**Comment by xaur** [comment]
@cball321 out of curiosity what was the VSP?

**Comment by cball321** [comment]
Sorry, I do not remember.   I didn't keep the records I should have.

**Comment by cball321** [comment]
I did get this resolved with the help of DCR support.
